Sources/ImageFetcher/ImageFetcher.swift

```diff
@@ -39,122 +39,105 @@ extension NSImage {
+        self.imageProcessor = imageProcessor
     }
 }
 
 // MARK: - Public API Methods
 public extension ImageFetcher {
-    /// Builds a `ImageLoaderTask`. If the result of the image configuration is cached, the task will be returned immediately. Otherwise a download operation will be kicked off.
+    /// Builds a `Task` to load an image for the given url.
     /// - Parameter url: The url of the image to be downloaded.
-    /// - Returns: An instance of `ImageLoaderTask`. Be sure to check `result` before adding a handler.
-    func task(_ url: URL) async -> ImageFetcherTask {
+    /// - Returns: The parent task of the image loading operation.
+    func task(_ url: URL) async -> Task<ImageSource, Error>  {
         await task(ImageConfiguration(url: url))
     }
 
-    /// Builds a `ImageLoaderTask`. If the result of the image configuration is cached, the task will be returned immediately. Otherwise a download operation will be kicked off.
+    /// Builds a `Task` to download the given image configuration.
     /// - Parameter imageConfiguration: The configuation of the image to be downloaded.
-    /// - Returns: An instance of `ImageLoaderTask`. Be sure to check `result` before adding a handler.
-    func task(_ imageConfiguration: ImageConfiguration) async -> ImageFetcherTask {
-        await withUnsafeContinuation { continuation in
-            workerQueue.sync {
-                // if data is cached, use it, else use `DataOperation` to fetch image data
-                if let cachedData = try? cache.syncData(imageConfiguration.key),
-                   let image = Image(data: cachedData)?.decompressed() {
-                    continuation.resume(
-                        returning: ImageFetcherTask(
-                            configuration: imageConfiguration,
-                            result: .success(.cached(image))))
-                } else {
-                    let operation = DataOperation(request: URLRequest(url: imageConfiguration.url), session: session)
-                    operation.name = imageConfiguration.key
-
-                    let task = ImageFetcherTask(configuration: imageConfiguration, operation: operation)
-                    operation.completionBlock = completion(task: task)
-                    queue.addOperation(operation)
-
-                    continuation.resume(returning: task)
-                }
+    /// - Returns: The parent task of the image loading operation.
+    func task(_ imageConfiguration: ImageConfiguration) async -> Task<ImageSource, Error> {
+        if let existingTask = getTask(imageConfiguration) {
+            return existingTask
+        } else if let cachedData = try? await cache.data(imageConfiguration.key) {
+            let decompressTask = Task(priority: imageConfiguration.priority) {
+                try await decompress(cachedData, imageConfiguration: imageConfiguration)
+            }
+
+            insertTask(decompressTask, key: imageConfiguration)
```

Comment:
This might need to be reworked as it is possible that a task with the same key has been inserted between the call to `getTask(_:)` and now, though I'm not sure how bad that would really be. Calls to `ImageFetcher.cancel(_:)` would fail to cancel the first task, and, assuming it completed first, the first task would remove the second from `tasks` when it completes, preventing cancellation of the second.